app/src/main/java/com/example/tasky/common/data/CalendarHelperImpl.kt

```diff
@@ -0,0 +1,28 @@
+package com.example.tasky.common.data
+
+import com.example.tasky.common.domain.CalendarHelper
+import com.example.tasky.feature_agenda.presentation.model.CalendarDay
+import java.time.LocalDate
+import java.util.Locale
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class CalendarHelperImpl @Inject constructor() : CalendarHelper {
+    override fun getCalendarDaysForMonth(year: Int, month: Int): List<CalendarDay> {
+        val firstDayOfMonth = LocalDate.of(year, month, 1)
+        val daysInMonth = firstDayOfMonth.lengthOfMonth()
+
+        val calendarDays = mutableListOf<CalendarDay>()
+        for (date in 1..daysInMonth) {
+            val currentDate = firstDayOfMonth.withDayOfMonth(date)
+            val dayOfWeek = currentDate.dayOfWeek.getDisplayName(
+                java.time.format.TextStyle.NARROW,
+                Locale.getDefault()
+            )
+            calendarDays.add(CalendarDay(dayOfWeek, date))
+        }
+
+        return calendarDays
+    }
+}
```

Comment:
What made you feel like this belongs in the data layer? I would argue that date formatting is something needed only for the presentation layer to display something, so could that be the better place? You also don't need the abstraction in that case.